Expensify/App
# Create Currency Modal, with geolocation and Location permissions

<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1970

### Tests


1. Go to /iou/split on web. Verify that the currency is based on your location.
2. Click on the currency, a list with all currencies should show.
3. Search for a currency, or scroll down to your desired currency 
4. Click on the currency to select and click on the Confirm button the confirm selection
5. Verify that the selected currency shows now in the amount page
6. Refresh the page, and verify that the same currency shows in the amount page

### QA Steps


### Tested On


### Screenshots


#### Web

<img width="1425" alt="Screenshot 2021-03-27 at 15 03 41" src="https://user-images.githubusercontent.com/30867740/112727434-91b76800-8f4a-11eb-8faa-12df727ad9fe.png">
<img width="1433" alt="Screenshot 2021-03-27 at 15 05 30" src="https://user-images.githubusercontent.com/30867740/112727437-9419c200-8f4a-11eb-9350-b612d2829c17.png">
<img width="1440" alt="Screenshot 2021-03-27 at 15 05 39" src="https://user-images.githubusercontent.com/30867740/112727438-954aef00-8f4a-11eb-978f-dc4a99ef2d52.png">
<img width="1431" alt="Screenshot 2021-03-27 at 15 05 47" src="https://user-images.githubusercontent.com/30867740/112727440-967c1c00-8f4a-11eb-94f8-97732edcf4b5.png">


#### Mobile Web


#### Desktop


#### iOS


https://user-images.githubusercontent.com/30867740/112727445-a0058400-8f4a-11eb-89ff-efbddf283607.mov


#### Android

<img width="321" alt="Screenshot 2021-03-27 at 15 28 28" src="https://user-images.githubusercontent.com/30867740/112727462-b7dd0800-8f4a-11eb-98a0-f01067ffd7cd.png">
<img width="308" alt="Screenshot 2021-03-27 at 15 28 39" src="https://user-images.githubusercontent.com/30867740/112727463-b9a6cb80-8f4a-11eb-9400-8628d79b573e.png">
<img width="324" alt="Screenshot 2021-03-27 at 15 28 44" src="https://user-images.githubusercontent.com/30867740/112727467-ba3f6200-8f4a-11eb-82ee-08f5a51e8d76.png">
<img width="317" alt="Screenshot 2021-03-27 at 15 28 51" src="https://user-images.githubusercontent.com/30867740/112727468-bad7f880-8f4a-11eb-9ffa-b2ed2f39a943.png">
